File: domain/src/main/scala/net/kemitix/thorp/domain/SizeTranslation.scala
package net.kemitix.thorp.domain

object SizeTranslation {

  val kbLimit: Long = 10240L
  val mbLimit: Long = kbLimit * 1024
  val gbLimit: Long = mbLimit * 1024

  def sizeInEnglish(length: Long): String =
    length.toDouble match {
      case bytes if bytes > gbLimit => f"${bytes / 1024 / 1024 / 1024}%.3fGb"
      case bytes if bytes > mbLimit => f"${bytes / 1024 / 1024}%.2fMb"
      case bytes if bytes > kbLimit => f"${bytes / 1024}%.0fKb"
      case bytes                    => s"${length}b"
    }

}

File: domain/src/main/java/net/kemitix/thorp/domain/SizeTranslation.java
package net.kemitix.thorp.domain;

public class SizeTranslation {
    static long kbLimit = 10240L;
    static long mbLimit = kbLimit * 1024;
    static long gbLimit = mbLimit * 1024;
    public static String sizeInEnglish(long length) {
        double bytes = length;
        if (length > gbLimit) {
            return String.format("%.3fGb", bytes / 1024 / 1024 / 1024);
        }
        if (length > mbLimit) {
            return String.format("%.2fMb", bytes / 1024 / 1024);
        }
        if (length > kbLimit) {
            return String.format("%.0fKb", bytes / 1024);
        }
        return String.format("%db", length);
    }
}
